File: 김현경/오픈채팅방.py
def solution(record):
    answer = []
    user_dict = {}
    
    for r in record:
        command = r.split()
        
        if command[0] == 'Enter' or command[0] == 'Change':
            user_dict[command[1]] = command[2]
        
    
    for r in record:
        command = r.split()
        
        if command[0] == 'Enter':
            answer.append(f'{user_dict[command[1]]}님이 들어왔습니다.')
        elif command[0] == 'Leave':
            answer.append(f'{user_dict[command[1]]}님이 나갔습니다.')

    return answer

File: 조성원/오픈채팅방.js
const solution = (record) => {
  const answer = [];
  const users = new Map();

  for (const string of record) {
    const data = string.split(" ");

    switch (data[0]) {
      case "Enter":
        users.set(data[1], data[2]);
        break;
      case "Change":
        users.set(data[1], data[2]);
        break;
      default:
        break;
    }
  }

  for (const string of record) {
    const data = string.split(" ");

    switch (data[0]) {
      case "Enter":
        answer.push(`${users.get(data[1])}님이 들어왔습니다.`);
        break;
      case "Leave":
        answer.push(`${users.get(data[1])}님이 나갔습니다.`);
        break;
      default:
        break;
    }
  }

  return answer;
};
